### PR TITLE
skip install ofed driver when the image is hpc type

### PR DESCRIPTION
**Pull request by LiliDeng**

### CODE REVIEW

File: lisa/features/infiniband.py [hunk 1]
```diff
@@ -171,7 +171,34 @@ def get_pkey(self) -> str:
...
+        sed = node.tools[Sed]
+        sed.substitute(
+            regexp="SELINUX=enforcing",
+            replacement="SELINUX=disabled",
+            file="/etc/selinux/config",
+            sudo=True,
+        )
+
+        if isinstance(node.os, Ubuntu) and node.os.information.version > "18.4.0":
+            node.tools[Sed].substitute(
```
Comment (squirrelsc) [inline lisa/features/infiniband.py:199]: Add comments above to explain the purpose.

Comment (LiliDeng) [reply lisa/features/infiniband.py:199]: updated

File: microsoft/testsuites/hpc/infinibandsuite.py [hunk 2]
```diff
@@ -433,20 +433,42 @@ def verify_ibm_mpi(self, environment: Environment, log: Logger) -> None:
         server_ssh.add_known_host(client_ip)
         client_ssh.add_known_host(server_ip)
 
+        if server_node.tools[Ls].path_exists("/usr/share/modules/init/bash"):
```
Comment (squirrelsc) [inline microsoft/testsuites/hpc/infinibandsuite.py]: why need to check this bash? what's different if it has or not? If it's really useful, add a comment above.

Comment (LiliDeng) [reply microsoft/testsuites/hpc/infinibandsuite.py]: it indicates a tool module has been installed, now I add is_hpc_image field.

**Review by squirrelsc — APPROVED** [review]

**Pull request merged**